department-of-veterans-affairs/caseflow
# Pamatya/appeals 36383

Resolves [APPEALS-36383](https://jira.devops.va.gov/browse/APPEALS-36383)

# Description
Changed how the seeds data were getting imported for vha_change_history. Instead of relying on traits, created methods in main seed file to create seed data.

## Acceptance Criteria
- [ ] Added issues with no decision date when there is a decision date on the request issue and currently no added_decision_date events associated with it. 
- [ ] A large amount of the tasks have a days waiting of 0 insted of a variable amount of time which would be nice. Setting the tasks.assigned_at time does this
- [ ]  The :with_update_users trait is creating a request_issues_updates with brand new request_issues in the before and after which does not really align with what we want. -> this has been removed. 
- [ ]  The cancelled tasks do not have a request issues updates to remove the one issue and cancel it.
- [ ] The completed tasks are missing a status event of Claim - In Progress because the status update time is within two seconds of object creation so the parser assumes
## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
2. Run the seeds data using this code `bundle exec rails runner scripts/run_seed_data_file.rb VhaChangeHistory`
3.  multiple new seeds data will be created in decision reviews tab. 
4. Go To generate Report page and generate the CSV.  **(NOTE: Use TYLERB/APPEALS-30892 branch to generate csv)**
5. Check if all the event types has been created for all the seeds data. 


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
